app/controllers/problems_controller.rb

```diff
@@ -0,0 +1,74 @@
+class ProblemsController < ApplicationController
+  before_action :set_problem, only: %I[show edit update destroy]
+  before_action :authenticate_user!, only: %I[new edit create destroy]
+
+  # GET /problems
+  # GET /problems.json
```

Comment:
Pls remove comment before commit, we can understand such simple help ourselves. But in case you doing complex algorithm that can not be simplified - in this case comments are good idea